Services/Simulations.cs

```diff
@@ -209,8 +212,10 @@ public async Task<Models.Simulation> UpsertAsync(Models.Simulation simulation)
 
                 if (simulation.ETag != existingSimulation.ETag)
                 {
-                    this.log.Error("Invalid ETag. Running simulation ETag is:'", () => new { simulation });
-                    throw new ResourceOutOfDateException("Invalid ETag. Running simulation ETag is:'" + simulation.ETag + "'.");
+                    var msg = "Invalid ETag. Running simulation ETag is:'";
+                    this.log.Error(msg, () => new { simulation });
+                    this.diagnosticsLogger.LogServiceErrorAsync(msg, new { simulation });
```

Comment:
logging the entire simulation object risks sending sensitive data. Please double check this with @hathind-ms and see if we can limit the data sent. We should check the entire codebase for the same, making sure this is not sending sensitive and unneeded data